cabinets/plugins.py

```diff
@@ -0,0 +1,75 @@
+import importlib
+import importlib.util
+import os
+import pkgutil
+import sys
+import inspect
+
+import cabinets.cabinet
+import cabinets.parser
+from cabinets.logger import info
+
+
+class CabinetsPluginError(Exception):
+    pass
+
+
+def discover(path, prefix=''):
+    plugins = pkgutil.iter_modules(path, prefix)
+    modules = set()
+    for _, name, _ in plugins:
+        module = importlib.import_module(name)
+        modules.add(module)
+    return modules
+
+
+def load_protocols(cls, protocols: dict):
+    if not cls._protocols:
+        raise CabinetsPluginError(f'No extensions registered to \'{cls.__name__}\'')
```

Comment:
Just to standardize with the rest of the package quoting conventions
```suggestion
        raise CabinetsPluginError(f"No extensions registered to '{cls.__name__}'")
```

Reply:
I hate your non-escaped quotes but whatever